src/db.ts

```diff
@@ -353,23 +352,37 @@ export abstract class Database {
     this.projects = new Map();
 
     this.queries.insertMeasurementBatchedN.text = `INSERT INTO Measurement
-         (runId, trialId, invocation, iteration, criterion, value)
-       VALUES ${this.generateBatchInsert(Database.batchN, 6)}
+         (runId, trialId, invocation, criterion, value)
+       VALUES ${this.generateBatchInsert(Database.batchN, 5)}
        ON CONFLICT DO NOTHING`;
 
     this.timelineUpdater = new SingleRequestOnly(async () => {
       return this.performTimelineUpdate();
     });
   }
 
+  
   private generateBatchInsert(numTuples: number, sizeTuples: number) {
     const nums: string[] = [];
     for (let i = 0; i < numTuples; i += 1) {
       const tupleNums: number[] = [];
       for (let j = 1; j <= sizeTuples; j += 1) {
         tupleNums.push(i * sizeTuples + j);
       }
-      nums.push('($' + tupleNums.join(', $') + ')');
+      // i am sure all of this can be done better and much tidyer
```

Comment:
Using `.join` as before doesn't work here?
How does the resulting string need to look like?

Reply:
[
  "( $1, $2, $3, $4, ARRAY [$5] )",
  "( $6, $7, $8, $9, ARRAY [$10] )",
  "( $11, $12, $13, $14, ARRAY [$15] )",
  "( $16, $17, $18, $19, ARRAY [$20] )",
  "( $21, $22, $23, $24, ARRAY [$25] )",
  "( $26, $27, $28, $29, ARRAY [$30] )",
  "( $31, $32, $33, $34, ARRAY [$35] )",
  "( $36, $37, $38, $39, ARRAY [$40] )",
  "( $41, $42, $43, $44, ARRAY [$45] )",
  "( $46, $47, $48, $49, ARRAY [$50] )",
  "( $51, $52, $53, $54, ARRAY [$55] )",
  "( $56, $57, $58, $59, ARRAY [$60] )",
  "( $61, $62, $63, $64, ARRAY [$65] )",
  "( $66, $67, $68, $69, ARRAY [$70] )",
  "( $71, $72, $73, $74, ARRAY [$75] )",
  "( $76, $77, $78, $79, ARRAY [$80] )",
  "( $81, $82, $83, $84, ARRAY [$85] )",
  "( $86, $87, $88, $89, ARRAY [$90] )",
  "( $91, $92, $93, $94, ARRAY [$95] )",
  "( $96, $97, $98, $99, ARRAY [$100] )",
  "( $101, $102, $103, $104, ARRAY [$105] )",
  "( $106, $107, $108, $109, ARRAY [$110] )",
  "( $111, $112, $113, $114, ARRAY [$115] )",
  "( $116, $117, $118, $119, ARRAY [$120] )",
  "( $121, $122, $123, $124, ARRAY [$125] )",
  "( $126, $127, $128, $129, ARRAY [$130] )",
  "( $131, $132, $133, $134, ARRAY [$135] )",
  "( $136, $137, $138, $139, ARRAY [$140] )",
  "( $141, $142, $143, $144, ARRAY [$145] )",
  "( $146, $147, $148, $149, ARRAY [$150] )",
  "( $151, $152, $153, $154, ARRAY [$155] )",
  "( $156, $157, $158, $159, ARRAY [$160] )",
  "( $161, $162, $163, $164, ARRAY [$165] )",
  "( $166, $167, $168, $169, ARRAY [$170] )",
  "( $171, $172, $173, $174, ARRAY [$175] )",
  "( $176, $177, $178, $179, ARRAY [$180] )",
  "( $181, $182, $183, $184, ARRAY [$185] )",
  "( $186, $187, $188, $189, ARRAY [$190] )",
  "( $191, $192, $193, $194, ARRAY [$195] )",
  "( $196, $197, $198, $199, ARRAY [$200] )",
  "( $201, $202, $203, $204, ARRAY [$205] )",
  "( $206, $207, $208, $209, ARRAY [$210] )",
  "( $211, $212, $213, $214, ARRAY [$215] )",
  "( $216, $217, $218, $219, ARRAY [$220] )",
  "( $221, $222, $223, $224, ARRAY [$225] )",
  "( $226, $227, $228, $229, ARRAY [$230] )",
  "( $231, $232, $233, $234, ARRAY [$235] )",
  "( $236, $237, $238, $239, ARRAY [$240] )",
  "( $241, $242, $243, $244, ARRAY [$245] )",
  "( $246, $247, $248, $249, ARRAY [$250] )",
]

this is now the resulting string

Reply:
ok, so, you are actually using parameterization already.

May that explain the real vs text error?
Could it be that the things in the array are not of the right type? perhaps the array you put in contains just strings?

Reply:
i have removed all ARRAY []